elyra/pipeline/validate.py

```diff
@@ -431,30 +430,36 @@ def _validate_environmental_variables(self, node_id: str, node_label: str,
                                        "propertyName": 'env_vars',
                                        "value": env_var})
 
-    def _validate_label(self, node_id: str, node_label: str, response: ValidationResponse) -> None:
+    def _validate_label(self, node_id: str, filename: str, node_label: str, response: ValidationResponse) -> None:
         """
         KFP specific check for the label name when constructing the node operation using dsl
         :param node_id: the unique ID of the node
+        :param filename: the name of the file with or without a relative path prefix
         :param node_label: the given node name or user customized name/label of the node
         :param response: ValidationResponse containing the issue list to be updated
         """
         label_name_max_length = 63
-        label_regex = re.compile('^[a-z0-9]([-a-z0-9]{0,62}[a-z0-9])?')
+        label_regex = re.compile('^[a-z0-9]([-_a-z0-9]{0,62}[a-z0-9])?')
         matched = label_regex.search(node_label)
 
-        if len(node_label) > label_name_max_length:
-            response.add_message(severity=ValidationSeverity.Error,
+        if os.path.basename(filename) == node_label and len(os.path.basename(filename)) > label_name_max_length or \
+                len(node_label) > label_name_max_length:
```

Comment:
This is a very confusing statement and I'd like to ask why we need to check against the `filename` at all?  Is there some way to adjust this issue.  The issue is equivalent to the following statement...
```python
if os.path.basename(filename) == node_label and len(node_label) > label_name_max_length or \
                len(node_label) > label_name_max_length:
```
which is equivalent to:
```python
if len(node_label) > label_name_max_length:
```
so I'm not sure what the `filename` check is really driving at.